### PR TITLE
Refactor/dedicated/server

### PR DESCRIPTION
**Pull request by marie-j**
Estimated prod date : 📆 08/07/2020

| Question         | Answer
| ---------------- | ---
| Branch?          | `develop`
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | MANAGER-4820
| License          | BSD 3-Clause

## Note for prodding
You can get rid of the sws-dedicated-server-tasks 2API once this is prodded

## Related
- [x] TRANS-25965

### CODE REVIEW

**Comment by marie-j** [comment]
Reopening https://github.com/ovh/manager/pull/2899

File: packages/manager/apps/dedicated/client/app/dedicated/server/dashboard/advanced-features/sgx/manage/manage.service.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import indexOf from 'lodash/indexOf';
+import snakeCase from 'lodash/snakeCase';
+
+import { DISPLAYING_ORDER } from './manage.constants';
+
+export default class {
```
Comment (JeremyDec) [inline packages/manager/apps/dedicated/client/app/dedicated/server/dashboard/advanced-features/sgx/manage/manage.service.js:6]: **suggestion**: This service should be a plain object, there are only static methods in it.

File: packages/manager/apps/dedicated/client/app/dedicated/server/dashboard/advanced-features/sgx/sgx.service.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import isString from 'lodash/isString';
+import toUpper from 'lodash/toUpper';
+
+import { DOCUMENTATION_BY_SUB } from './sgx.constants';
+
+export default class {
```
Comment (JeremyDec) [inline packages/manager/apps/dedicated/client/app/dedicated/server/dashboard/advanced-features/sgx/sgx.service.js:6]: **suggestion**: This service should be a plain object, there are only static methods in it.

File: packages/manager/apps/dedicated/client/app/dedicated/server/server.service.js [hunk 3]
```diff
@@ -0,0 +1,2406 @@
...
+import head from 'lodash/head';
+import indexOf from 'lodash/indexOf';
+import map from 'lodash/map';
+import parseInt from 'lodash/parseInt';
+import set from 'lodash/set';
+import snakeCase from 'lodash/snakeCase';
+import sortBy from 'lodash/sortBy';
+import uniq from 'lodash/uniq';
+
+export default class ServerF {
```
Comment (JeremyDec) [inline packages/manager/apps/dedicated/client/app/dedicated/server/server.service.js:15]: **nit**: weird class name no ?

Comment (marie-j) [reply packages/manager/apps/dedicated/client/app/dedicated/server/server.service.js:15]: @JeremyDec I think it was done that way due to a `Server` service already existing

File: packages/manager/apps/dedicated/client/app/dedicated/server/dashboard/advanced-features/sgx/manage/translations/Messages_fr_FR.json [hunk 4]
```diff
@@ -0,0 +1,19 @@
+{
+  "dedicated_server_dashboard_advanced_features_sgx_manage_back_button_dashboard": "Retour à l'état du serveur",
+  "dedicated_server_dashboard_advanced_features_sgx_manage_back_button_presentation": "Retour à la présentation de la fonctionnalité",
+  "dedicated_server_dashboard_advanced_features_sgx_manage_header_disabled": "Gestion de votre fonctionnalité SGX",
+  "dedicated_server_dashboard_advanced_features_sgx_manage_header_enabled": "Modification de votre fonctionnalité SGX",
+  "dedicated_server_dashboard_advanced_features_sgx_manage_header_software controlled": "Modification de votre fonctionnalité SGX",
```
[the remark this answers is not in the document]
Comment (marie-j) [reply packages/manager/apps/dedicated/client/app/dedicated/server/dashboard/advanced-features/sgx/manage/translations/Messages_fr_FR.json:6]: I'll have a look for the usage, but looks like it is to map some statuses

File: packages/manager/apps/dedicated/client/app/dedicated/server/dashboard/advanced-features/sgx/status/status.service.js [hunk 5]
```diff
@@ -0,0 +1,56 @@
+import includes from 'lodash/includes';
+import keys from 'lodash/keys';
+import values from 'lodash/values';
+
+import { STATUS } from '../sgx.constants';
+
+export default class {
```
[the remark this answers is not in the document]
Comment (marie-j) [reply packages/manager/apps/dedicated/client/app/dedicated/server/dashboard/advanced-features/sgx/status/status.service.js:7]: @JeremyDec What do you think about just exporting utility functions ?

**Comment by marie-j** [comment]
@JeremyDec Thank you for your remarks, due to the fact that it is in a "quality check ok" state and that nothing looks like a blocker, I'm taking note of your suggestions to apply them later on (Internal reference to come)

File: packages/manager/apps/dedicated/client/app/dedicated/server/dashboard/advanced-features/sgx/status/status.service.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (JeremyDec) [reply packages/manager/apps/dedicated/client/app/dedicated/server/dashboard/advanced-features/sgx/status/status.service.js:7]: Why not :). If the service is not used as a angular injected service we could do so

**Pull request merged**